commands/init.go

```diff
@@ -5,8 +5,22 @@ import (
 
 	"github.com/mattermost/mattermost-server/model"
 	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
 )
 
+func withClient(fn func(c *model.Client4, cmd *cobra.Command, args []string) error) func(cmd *cobra.Command, args []string) error {
+	c, err := InitClient()
+	if err != nil {
+		return func(cmd *cobra.Command, args []string) error {
+			return err
+		}
+	}
+
+	return func(cmd *cobra.Command, args []string) error {
+		return fn(c, cmd, args)
+	}
```

Comment:
This is going to execute the initialization of the client for each method created, I think I prefer something like>

```suggestion
return func(cmd *cobra.Command, args []string) error {
	c, err := InitClient()
	if err != nil {
		return err
	}
	return fn(c, cmd, args)
}
```

Reply:
Great catch!! Will change :)